src/hats_import/verification/arguments.py

```diff
@@ -3,43 +3,51 @@
-    """In-memory representation of a catalog. If not provided, it will be loaded
-    from the input_catalog_path."""
-
-    ## Verification options
-    field_distribution_cols: List[str] = field(default_factory=list)
-    """List of fields to get the overall distribution for. e.g. ["ra", "dec"].
-    Should be valid columns in the parquet files."""
-
-    def __post_init__(self):
-        self._check_arguments()
-
-    def _check_arguments(self):
-        super()._check_arguments()
-        if not self.input_catalog_path and not self.input_catalog:
-            raise ValueError("input catalog is required (either input_catalog_path or input_catalog)")
-        if not self.input_catalog:
-            if not is_valid_catalog(self.input_catalog_path):
-                raise ValueError("input_catalog_path not a valid catalog")
-            self.input_catalog = read_hats(catalog_path=self.input_catalog_path)
-        if not self.input_catalog_path:
-            self.input_catalog_path = self.input_catalog.catalog_path
+
+@dataclass(kw_only=True)
+class VerificationArguments:
+    """Container for verification arguments."""
+
+    input_catalog_path: UPath = field()
+    """Path to an existing catalog that will be inspected. This must be a directory
+    containing (at least) the hats ancillary files and a 'dataset/' directory
+    containing the parquet dataset. Can be supplied as a string or path object."""
+    output_path: UPath = field()
+    """Directory where the verification report should be written.
+     Can be supplied as a string or path object."""
+    output_filename: str = field(default="verifier_results.csv")
+    """Filename for the verification report."""
+    truth_total_rows: int | None = field(default=None)
+    """Total number of rows expected in this catalog."""
+    truth_schema: UPath | None = field(default=None)
+    """Path to a parquet file or dataset containing the expected schema. If None (default),
+    the catalog's _common_metadata file will be used. This schema will be used to verify
+    all non-hats columns and (optionally) the file-level metadata. Can be supplied as a
+    string or path object."""
+
+    @property
+    def input_dataset_path(self) -> UPath:
+        """Path to the directory under `input_catalog_path` that contains the parquet dataset."""
+        return file_io.append_paths_to_pointer(self.input_catalog_path, hats.io.paths.DATASET_DIR)
+
+    @property
+    def output_file_path(self) -> UPath:
+        """Path to the output file (`output_path` / `output_filename`)."""
+        return file_io.append_paths_to_pointer(self.output_path, self.output_filename)
+
+    def __post_init__(self) -> None:
+        self.input_catalog_path = file_io.append_paths_to_pointer(self.input_catalog_path)
+        if not self.input_catalog_path.is_dir():
+            raise ValueError("input_catalog_path must be an existing directory")
+
+        self.output_path = file_io.append_paths_to_pointer(self.output_path)
```

Comment:
```suggestion
        self.output_path = file_io.get_upath(self.output_path)
```

Reply:
We should have a test for this in our `hats-cloudtests` repo, as that's where we confirm that non-local storage options are propagated via end-to-end tests. I can take on that task.